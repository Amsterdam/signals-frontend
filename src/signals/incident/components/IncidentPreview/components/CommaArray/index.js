import React from 'react';
import PropTypes from 'prop-types';
import PlainText from '../PlainText';

const CommaArray = ({ value, ...props }) => <PlainText value={value.join(', ')} {...props} />;

CommaArray.propTypes = {
  label: PropTypes.string,
  value: PropTypes.arrayOf(PropTypes.string).isRequired,
};

export default CommaArray;
