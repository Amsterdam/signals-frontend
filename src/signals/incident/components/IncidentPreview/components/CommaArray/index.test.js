import React from 'react';
import { shallow } from 'enzyme/build';
import CommaArray from './index';

describe('the comma array preview component', () => {
  it('should comma separate values', () => {
    const wrapper = shallow(<CommaArray label="foo" value={['1', '2']} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should handle empty array', () => {
    const wrapper = shallow(<CommaArray label="foo" value={[]} />);
    expect(wrapper).toMatchSnapshot();
  });
});
