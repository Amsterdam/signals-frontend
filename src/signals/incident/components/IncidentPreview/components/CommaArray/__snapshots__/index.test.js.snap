// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the comma array preview component should comma separate values 1`] = `
<PlainText
  label="foo"
  value="1, 2"
/>
`;

exports[`the comma array preview component should handle empty array 1`] = `
<PlainText
  label="foo"
  value=""
/>
`;
